submissions/smillims/oop-exercise/index.js

```diff
@@ -0,0 +1,85 @@
+/* 
+  Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+*/
+
+class Inhabitant {
+  constructor(surname, species, gender, name, legs){
+    this.surname = surname;
+    this.species = species;
+    this.gender = gender;
+    this.name = name;
+    this.legs = legs;
+  }
+  toString(){
+    return [`Hi, I'm ` + this.species,
+    `My name is ` + this.name,
+    `My gender is ` + this.gender,
+    `I have ` + this.legs + ` legs`, ``].join(', ')
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(surname, name, gender, hands = 2) {
+    super(surname, 'human', gender, name, 2);
+    this.hands = hands;
+  }
+}
+
+class Man extends Human {
+  constructor(name, surname) {
+    super(surname, name, 'male')
+  }
+  toString() {
+    return super.toString() + [`I have ` + this.hands + ` hands`, 
+                            ` My surname is ` + this.surname]
+  }
+}
+
+class Woman extends Human {
+  constructor(name, surname) {
+    super(surname, name, 'female')
+  }
+  toString() {
+    return super.toString() + [`I have ` + this.hands + ` hands`, 
+                            ` My surname is ` + this.surname]
+  }
+}
+
+class Cat extends Inhabitant {
+  constructor(surname, name, gender) {
+    super(surname, 'cat', gender, name, '4')
+  }
+}
+
+class SheCat extends Cat {
+  constructor(name, surname) {
+    super(surname, name, 'female')
+  }
+  toString() {
+    return super.toString() + [`My surname is ` + this.surname]
```

Comment:
Easily avoidable code duplication. Perhaps, you won't need this method  at all when `surname` is handled properly.
The class will inherit the method from its parent (and its parent if immediate parent doesn't have the method overridden, etc)